src/javascript/ContentEditor/SelectorTypes/ChoiceTree/ChoiceTree.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FieldPropTypes} from '~/ContentEditor/ContentEditor.proptypes';
+import {useQuery} from '@apollo/client';
+import {GetTreeEntries} from './choiceTree.gql-queries';
+import {useTranslation} from 'react-i18next';
+import {LoaderOverlay} from '~/ContentEditor/DesignSystem/LoaderOverlay';
+import {Dropdown} from '@jahia/moonstone';
+import {choiceTreeAdapter} from "./choiceTree.adapter";
+
+export const ChoiceTree = ({field, value, id, editorContext, onChange, onBlur}) => {
+    const {t} = useTranslation('jcontent');
+    const {data, error, loading} = useQuery(GetTreeEntries, {
+        variables: {
+            types:  field.selectorOptions?.find(option => option.name === 'types')?.value?.split(','),
+            path: field.selectorOptions?.find(option => option.name === 'rootPath')?.value,
+            language: editorContext.lang
+        }
+    });
+
+    const handleClear = () => {
+        if (field.multiple) {
+            onChange([]);
+        } else {
+            onChange(null);
+        }
+    };
+
+    const handleChange = (_, selectedValue) => {
+        if (field.multiple) {
+            const prev = value || [];
+            onChange(prev.indexOf(selectedValue.value) > -1 ? prev.filter(i => i !== selectedValue.value) : [...prev, selectedValue.value]);
+        } else {
+            onChange(selectedValue.value);
+        }
+    };
+
+    if (!field.multiple) {
+        console.warn('ChoiceTree selector is not supporting single value');
+        return <></>;
+    }
+
+    if (error) {
+        const message = t(
+            'jcontent:label.contentEditor.error.queryingContent',
+            {details: `${field.selectorOptions?.find(option => option.name === 'rootPath')?.value} in ${editorContext.lang}`}
+        );
+        console.warn('unable to resolve choice tree configuration from selector options', field.selectorOptions, error);
+        return <>{message}</>;
+    }
+
+    if (loading) {
+        return <LoaderOverlay/>;
+    }
+
+    const tree = choiceTreeAdapter({
+        nodes: data.jcr.result.descendants.nodes,
+        parent: data.jcr.result,
+        selectedValues: value,
+        locale: editorContext.lang
+    });
+
+    const singleValue = field.multiple ? undefined : value;
+    const multipleValue = field.multiple ? (value || []) : undefined;
```

Comment:
I don't see why we have single vs multiple logic here, couple of line upper there is already:
```
if (!field.multiple) {
        console.warn('ChoiceTree selector is not supporting single value');
        return <></>;
}
```

Reply:
single logic has been preserved